xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-security/xwiki-platform-extension-security-index/src/main/resources/ApplicationResources.properties

```diff
@@ -18,6 +18,7 @@
 # 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 # ---------------------------------------------------------------------------
 
-extension.security.analysis.advice.upgradeFromEM=This extension can be upgraded from the extension manager.
-extension.security.analysis.advice.upgradeXWiki=Please upgrade XWiki.
-extension.security.analysis.advice.upgradeEnvironment=Please upgrade your servlet engine.
\ No newline at end of file
+extension.security.analysis.advice.upgradeFromEM=Once a new version with a fix is available, this extension can be upgraded from the extension manager.
+extension.security.analysis.advice.transitive=This extension is installed because it is required by another extension. It is not advised to manually upgrade it. Instead, you should upgrade the extension requiring it. Please see them listed below.
```

Comment:
> Please see them listed below.

You are only listing top level extension, right ? Not all backward dependencies ?

Reply:
Having link in this list would also make things a lot easier if that's possible in this kind of modal.